Add more required fields when users signup for better user tracking and reporting

hyi opened the pull request:
<!--

Please read, and add your text at the bottom of this message.

Thanks for contributing code to HydroShare. In order to maintain code quality and expedite this process, please assist the development team by making sure the following is present in this pull request.

For more information, see https://docs.google.com/document/d/1dzxqlZW5fKNEyQSeKiSFq-SmS-VOPCva95XXkBjPExs

-->

### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

<!-- Enter steps that a QA engineer, stakeholder, or user documentation writer would follow to test the positive or "successful" case of the functionality your code provides or fixes -->

- [ ] Automated Testing

<!-- Our Jenkins Instance is set up to automatically test every commit from a pull request. Code coverage must not decrease so new functionality or code paths added during a bug fix must have appropriate tests written. Every test must pass, including PEP8 code formatting tests. -->

- [ ] Sufficient User and Developer Documentation

<!-- Please email your positive test case lbrazil@cuahsi.org, who will make the decision regarding user documentation. -->

- [ ] Passing Jenkins Build

<!-- Our Jenkins Instance is set up to automatically test every commit from a pull request. Every test must pass, including PEP8 code formatting tests. -->

- [ ] Peer Code review and approval

<!-- This is the process by which a peer developer on the HydroShare team will read the changeset, provide feedback, and ultimately give a formal approval to the code before it passes PR status. -->

### Positive Test Case
1. [Enter positive test case here]

hyi commented:
@sblack-usu @Castronova I assigned both of you to code review this PR which addresses the issue #3838.

hydrocheck commented:
| METRIC | VALUE |
|:-----:|:-----:|
| NEW CODE SMELLS | 1 |
| NEW SECURITY HOTSPOTS | 0 |
| NEW VULNERABILITIES | 0 |
| NEW BUGS | 3 |
| NEW DUPLICATED LINES DENSITY | 0.0 |

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-3972](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-3972)

Castronova commented:
@hyi Is there a place where this can be tested, i.e. beta?

hyi commented:
@Castronova This work has been deployed to https://new-playground.hydroshare.org/sign-up 
You can test it out there while reviewing this PR.

sblack-usu commented on theme/templates/accounts/profile.html:
```diff
@@ -415,7 +414,7 @@ <h3 class="panel-title">
                                             <fieldset>
                                                 <label>State/Province *</label>
-                                                <input name="state" class="form-control form-required" {% if  profile.state %}value="{{ profile.state }}"{% else %}value="Unspecified"{% endif %}>
+                                                <input name="state" class="form-control form-required" value="{{ profile.state }}">
```
Should country be added to?

hyi replied:
@sblack-usu I have made a new commit to address this. Basically, for those existing users who have not specified country and state, they should be left empty (rather than None or unspecified) in the input fields on the profile page so that they have to select valid values in order to save updated profile. This will make sure no unspecified country or None state got saved to user profile any more. You can test it out on the new-playground.hydroshare.org.

sblack-usu commented on theme/templates/accounts/sign_up_form.html:
```diff
@@ -97,12 +148,24 @@
             var missingFields = 'Data missing for:';
             var missingFieldCount = 0;
             var j;
+            if($("#selectUserType").val() == 'Other' && $("#user-type-other  input").val() == '') {
```
any reason you can't add these fields to requiredFieldsToCheck?

sblack-usu replied:
oh, I see what's going on now.

hyi commented:
@sblack-usu Are you done with the code review for this PR? I see @Castronova approved the PR, but I want to make sure I addressed all your comments before merging it.

sblack-usu commented:
It needs a couple of tests around the changes.

hyi commented:
@sblack-usu The changes I made are all in front end. Can you be more specific on what tests you think that are needed?

sblack-usu commented:
I was thinking with create_account and possibly the form updates.  Not sure how much work it would be to wrap the form updates into a test but the create_account method should be straightforward.

hyi commented:
retest this, please

hyi commented:
@sblack-usu test has been added. Take a look when you have a chance.
